File: Phase-III-Ai-Chatbot/backend/test_auth.py
#!/usr/bin/env python3
"""
Test script to verify the authentication fix.
"""
import requests
import json
import uuid
from datetime import datetime, timedelta
from jose import jwt

# Test server URL
BASE_URL = "http://localhost:8000"

def create_test_token():
    """Create a valid test JWT token."""
    secret = "fallback_secret_for_development"  # Same as in the backend
    user_id = str(uuid.uuid4())

    payload = {
        "user_id": user_id,
        "exp": datetime.utcnow() + timedelta(hours=24),
        "iat": datetime.utcnow().timestamp(),
        "sub": "test@example.com"
    }

    token = jwt.encode(payload, secret, algorithm="HS256")
    return token, user_id

def test_unauthorized_request():
    """Test request without authentication."""
    print("Testing unauthorized request to /api/tasks/...")
    try:
        response = requests.get(f"{BASE_URL}/api/tasks/")
        print(f"Status Code: {response.status_code}")
        print(f"Response: {response.json()}")
        return response.status_code
    except Exception as e:
        print(f"Error: {e}")
        return None

def test_authorized_request():
    """Test request with proper authentication."""
    print("\nTesting authorized request to /api/tasks/...")

    # Create a valid token
    token, user_id = create_test_token()
    print(f"Created test token for user: {user_id}")

    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json"
    }

    try:
        response = requests.get(f"{BASE_URL}/api/tasks/", headers=headers)
        print(f"Status Code: {response.status_code}")
        if response.status_code == 200:
            print(f"Response: {response.json()}")
            print("✅ Authentication working correctly!")
        else:
            print(f"Response: {response.json()}")
            print("❌ Authentication may still have issues")
        return response.status_code
    except Exception as e:
        print(f"Error: {e}")
        return None

def test_health_endpoint():
    """Test the health endpoint which should not require authentication."""
    print("\nTesting health endpoint (should not require auth)...")
    try:
        response = requests.get(f"{BASE_URL}/health")
        print(f"Status Code: {response.status_code}")
        print(f"Response: {response.json()}")
        return response.status_code
    except Exception as e:
        print(f"Error: {e}")
        return None

if __name__ == "__main__":
    print("Testing Backend Authentication Fix\n")
    print("=" * 40)

    # Test health endpoint first
    test_health_endpoint()

    # Test unauthorized request
    test_unauthorized_request()

    # Test authorized request
    test_authorized_request()

    print("\n" + "=" * 40)
    print("Test completed!")

File: Phase-V/backend/test_auth.js
#!/usr/bin/env node
// Test script to verify authentication and task functionality

const BASE_URL = 'http://127.0.0.1:8000';

async function testAuthFlow() {
  console.log('Testing authentication flow...\n');

  // Test 1: Try to access protected endpoint without authentication
  console.log('1. Testing protected endpoint without authentication...');
  try {
    const response = await fetch(`${BASE_URL}/api/tasks/`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json'
      }
    });
    console.log(`   Response: ${response.status} ${response.statusText}`);
    if (response.status === 401) {
      console.log('   ✅ Correctly returns 401 without authentication');
    } else {
      console.log('   ❌ Expected 401, got different response');
    }
  } catch (error) {
    console.error('   ❌ Error testing protected endpoint:', error.message);
  }

  // Test 2: Try to access public endpoint
  console.log('\n2. Testing public endpoint...');
  try {
    const response = await fetch(`${BASE_URL}/health`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json'
      }
    });
    console.log(`   Response: ${response.status} ${response.statusText}`);
    if (response.status === 200) {
      console.log('   ✅ Health endpoint working correctly');
    } else {
      console.log('   ❌ Health endpoint not working as expected');
    }
  } catch (error) {
    console.error('   ❌ Error testing health endpoint:', error.message);
  }

  // Test 3: Try to access auth endpoints
  console.log('\n3. Testing authentication endpoints...');
  try {
    const response = await fetch(`${BASE_URL}/api/auth/login`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        email: 'test@example.com',
        password: 'password'
      })
    });
    console.log(`   Login endpoint response: ${response.status} ${response.statusText}`);

    // Test signup endpoint too
    const signupResponse = await fetch(`${BASE_URL}/api/auth/sign-up/email`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name: 'Test User',
        email: 'test@example.com',
        password: 'password123'
      })
    });
    console.log(`   Signup endpoint response: ${signupResponse.status} ${signupResponse.statusText}`);
  } catch (error) {
    console.error('   ❌ Error testing auth endpoints:', error.message);
  }

  console.log('\nTest completed. Backend authentication system is accessible.');
}

// Run the test
testAuthFlow().catch(console.error);